### PR TITLE
Removes SEXP, SYMBOL, and DECIMAL_ARBITRARY from PType and Datum

### PR DESCRIPTION
**Pull request by johnedquinn**
## Description
- Removes SEXP, SYMBOL, DECIMAL_ARBITRARY from PType.

## Testing
- All tests passing.
- For conformance tests, please see the report. The only failing tests are tests that are due to symbol/sexp.

## Follow-ups
- Remove/modify NUMERIC. Do we want to keep NUMERIC as SQL's NUMERIC? We might want to, since it is in the SQL Spec.

## License Information

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: partiql-eval/src/main/kotlin/org/partiql/eval/internal/operator/rel/RelOpExclude.kt [hunk 1]
```diff
@@ -173,7 +172,7 @@ internal class RelOpExclude(
             // apply exclusions to subtree
             var value = element
             // apply collection index exclusions at deeper levels for lists and sexps
```
Comment (alancai98) [inline partiql-eval/src/main/kotlin/org/partiql/eval/internal/operator/rel/RelOpExclude.kt:174]: Can remove reference to `sexps` here

```suggestion
            // apply collection index exclusions at deeper levels for lists
```

File: partiql-spi/src/main/kotlin/org/partiql/value/io/PartiQLValueIonReader.kt [hunk 2]
```diff
@@ -124,15 +122,7 @@ internal class PartiQLValueIonReader(
...
-            IonType.SYMBOL -> {
-                if (reader.isNullValue) {
-                    symbolValue(null, reader.typeAnnotations.toList())
-                } else {
-                    symbolValue(reader.stringValue(), reader.typeAnnotations.toList())
-                }
-            }
-
-            IonType.STRING -> {
+            IonType.STRING, IonType.SYMBOL -> {
```
Comment (alancai98) [inline partiql-spi/src/main/kotlin/org/partiql/value/io/PartiQLValueIonReader.kt:125]: For the `PartiQLValueIonReader` changes, what's the intended behavior for Ion types that we are not representing in PartiQL? Seems like this PR coerces `SYMBOL` to PartiQL string values and `SEXP`s to PartiQL list values. Should we give an error instead?

File: partiql-spi/src/main/kotlin/org/partiql/value/PartiQLValueType.kt [hunk 3]
```diff
@@ -70,16 +70,17 @@ public enum class PartiQLValueType {
     )
     public fun toPType(): PType {
         return when (this) {
-            DECIMAL, DECIMAL_ARBITRARY -> PType.decimal()
+            DECIMAL -> PType.decimal()
+            DECIMAL_ARBITRARY -> PType.decimal() // TODO: Figure out precision/scale implications.
```
Comment (alancai98) [inline partiql-spi/src/main/kotlin/org/partiql/value/PartiQLValueType.kt:74]: Should decimal_arb also give an error like you do for symbol and sexp?

File: partiql-spi/src/main/kotlin/org/partiql/spi/function/builtins/FnPlus.kt [hunk 4]
```diff
@@ -89,8 +89,8 @@ internal val Fn_PLUS__DECIMAL_ARBITRARY_DECIMAL_ARBITRARY__DECIMAL_ARBITRARY = F
     name = "plus",
     returns = PType.decimal(),
     parameters = arrayOf(
-        @Suppress("DEPRECATION") Parameter("lhs", PType.decimal()),
-        @Suppress("DEPRECATION") Parameter("rhs", PType.decimal()),
+        Parameter("lhs", PType.decimal()),
```
Comment (alancai98) [inline partiql-spi/src/main/kotlin/org/partiql/spi/function/builtins/FnPlus.kt:92]: `FN` name var for above uses DECIMAL_ARBITRARY, perhaps needs to get changed?

File: partiql-spi/src/main/kotlin/org/partiql/spi/function/builtins/FnLte.kt [hunk 5]
```diff
@@ -88,8 +88,8 @@ internal val Fn_LTE__DECIMAL_ARBITRARY_DECIMAL_ARBITRARY__BOOL = Function.static
     name = "lte",
     returns = PType.bool(),
     parameters = arrayOf(
-        @Suppress("DEPRECATION") Parameter("lhs", PType.decimal()),
-        @Suppress("DEPRECATION") Parameter("rhs", PType.decimal()),
+        Parameter("lhs", PType.decimal()),
```
Comment (alancai98) [inline partiql-spi/src/main/kotlin/org/partiql/spi/function/builtins/FnLte.kt:91]: Similarly, fn name variable uses DECIMAL_ARB. Perhaps should get changed?

Also applies for
- lt
- gt
- gte
- between
- may have some others, so probably need to do a check on Builtins.kt

File: partiql-spi/src/main/kotlin/org/partiql/spi/function/builtins/FnConcat.kt [hunk 6]
```diff
@@ -8,34 +8,45 @@ import org.partiql.spi.function.Parameter
 import org.partiql.spi.value.Datum
 import org.partiql.types.PType
 
-internal val Fn_CONCAT__STRING_STRING__STRING = Function.static(
-
+internal val Fn_CONCAT__CHAR_CHAR__CHAR = Function.static(
     name = "concat",
-    returns = PType.string(),
+    returns = PType.character(256), // TODO: Handle length
```
Comment (alancai98) [inline partiql-spi/src/main/kotlin/org/partiql/spi/function/builtins/FnConcat.kt:13]: Should these TODOs for length be handled in this PR or tracked somewhere else?

File: partiql-planner/src/test/kotlin/org/partiql/planner/util/Utils.kt [hunk 7]
```diff
@@ -32,6 +33,55 @@ val allSupportedType = StaticType.ALL_TYPES.filterNot {
...
+
+val allCharStringPType = setOf(
+    PType.character(256), // TODO: Length
+    PType.varchar(256), // TODO: Length
+    PType.string(),
+    PType.clob(Int.MAX_VALUE), // TODO: Length
+)
+
+val allBinaryPType = setOf(
+    PType.blob(Int.MAX_VALUE), // TODO: Length
```
Comment (alancai98) [inline partiql-planner/src/test/kotlin/org/partiql/planner/util/Utils.kt:58]: Will these TODOs for precision/length be addressed in this PR or are they tracked in some other issue?

File: partiql-spi/src/main/kotlin/org/partiql/spi/function/builtins/FnPlus.kt (hunk 4 above)
Comment (johnedquinn) [reply partiql-spi/src/main/kotlin/org/partiql/spi/function/builtins/FnPlus.kt:92]: Eventually, yes. Currently rewriting all of the functions. See #1647

File: partiql-spi/src/main/kotlin/org/partiql/spi/function/builtins/FnConcat.kt (hunk 6 above)
Comment (johnedquinn) [reply partiql-spi/src/main/kotlin/org/partiql/spi/function/builtins/FnConcat.kt:13]: The work for this has begun in #1647.

File: partiql-planner/src/test/kotlin/org/partiql/planner/util/Utils.kt (hunk 7 above)
Comment (johnedquinn) [reply partiql-planner/src/test/kotlin/org/partiql/planner/util/Utils.kt:58]: https://github.com/partiql/partiql-lang-kotlin/issues/1649

**Review by alancai98 — APPROVED** [review]
Nice, LGTM. May also want to update the PR title and commit messages to include that DECIMAL_ARBITRARY was removed.

**Pull request merged**